### PR TITLE
[CMK] Encryption configuration + New FHIR runtime configurations

### PR DESCRIPTION
**Pull request by fhibf**
New configuration classes to handle encryption.

## Description
Describe the changes in this PR.

## Related issues
Addresses [issue #].

## Testing
Describe how this change was tested.

## FHIR Team Checklist
- **Update the title** of the PR to be succinct and less than 65 characters
- **Add a milestone** to the PR for the sprint that it is merged (i.e. add S47)
- Tag the PR with the type of update: **Bug**, **Build**, **Dependencies**, **Enhancement**, **New-Feature** or **Documentation**
- Tag the PR with **Open source**, **Azure API for FHIR** (CosmosDB or common code) or **Azure Healthcare APIs** (SQL or common code) to specify where this change is intended to be released.
- Tag the PR with **Schema Version backward compatible** or **Schema Version backward incompatible** or **Schema Version unchanged** if this adds or updates Sql script which is/is not backward compatible with the code.
- [ ] CI is green before merge [Build Status](https://microsofthealthoss.visualstudio.com/FhirServer/_build/latest?definitionId=27&branchName=main) 
- Review [squash-merge requirements](https://github.com/microsoft/fhir-server/blob/master/SquashMergeRequirements.md)

### Semver Change ([docs](https://github.com/microsoft/fhir-server/blob/master/docs/Versioning.md))
Patch|Skip|Feature|Breaking (reason)

### CODE REVIEW

File: src/Microsoft.Health.Fhir.Core.UnitTests/Config/EncryptionConfigurationTests.cs [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+            {
+                CustomerManagedKeyEncryption = new CustomerManagedKeyEncryption()
+                {
+                    KeyEncryptionKeyIdentity = new EncryptionKeyIdentity()
+                    {
+                        FederatedClientId = "foo",
+                    },
+                },
+            };
+            Assert.True(encryption0.IsEncryptionSet());
```
Comment (mikaelweave) [inline src/Microsoft.Health.Fhir.Core.UnitTests/Config/EncryptionConfigurationTests.cs:31]: Is this a valid configuration? I thought the KeyUrl was always required.

File: src/Microsoft.Health.Fhir.Core.UnitTests/Config/EncryptionConfigurationTests.cs [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+            EncryptionConfiguration encryption1 = new EncryptionConfiguration()
+            {
+                CustomerManagedKeyEncryption = new CustomerManagedKeyEncryption()
+                {
+                    KeyEncryptionKeyUrl = new Uri("htts://fhir.com"),
```
Comment (mikaelweave) [inline src/Microsoft.Health.Fhir.Core.UnitTests/Config/EncryptionConfigurationTests.cs:37]: NIT: Should this be `https`?

**Review by mikaelweave — COMMENTED** [review]
A couple questions on this PR

File: src/Microsoft.Health.Fhir.Core.UnitTests/Config/EncryptionConfigurationTests.cs (hunk 1 above)
Comment (fhibf) [reply src/Microsoft.Health.Fhir.Core.UnitTests/Config/EncryptionConfigurationTests.cs:31]: I need to sync with you about this as well. 
From your tests, are both properties required? Or just the KeyUrl is required? 
What is the usage of the FederalClientId?

Comment (fhibf) [reply src/Microsoft.Health.Fhir.Core.UnitTests/Config/EncryptionConfigurationTests.cs]: I made FederatedClientId optional and KeyEncryptionKeyUrl required.

**Review by mikaelweave — APPROVED** [review]
Looks good (pending tests)! 🦾

**Pull request merged**